java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightJdbcFlightStreamResultSet.java

```diff
@@ -204,6 +259,21 @@ public boolean next() throws SQLException {
     }
   }
 
+  private void cleanUpResources() throws Exception {
+    if (flightStreamQueue != null) {
+      // flightStreamQueue should close currentFlightStream internally
+      flightStreamQueue.close();
+    } else if (currentFlightStream != null) {
+      // close is only called for currentFlightStream if there's no queue
+      currentFlightStream.close();
+    }
+
+    List<VectorSchemaRoot> roots = new LinkedList<>();
+    vectorSchemaRoots.drainTo(roots);
+    roots.forEach(AutoCloseables::closeNoChecked);
+    ofNullable(currentRoot).ifPresent(AutoCloseables::closeNoChecked);
```

Comment:
It would be preferable to do this all as one big AutoCloseables call if possible (also, is LinkedList strictly necessary here vs ArrayList?)